pkg/network/cluster_config.go

```diff
@@ -131,3 +135,19 @@ func StatusFromOperatorConfig(operConf *operv1.NetworkSpec, oldStatus *configv1.
 
 	return &status
 }
+
+// networkTypeRequiresHostPrefix checks whether the hostPrefix for the networkType needs to be
+// validated or not depending on if the networkType uses the hostPrefix.
+// It returns true if the networkType requires hostPrefix or
+// if the networkType that does not require a hostPrefix has a non-zero hostPrefix.
+func networkTypeRequiresHostPrefix(networkType string, hostPrefix uint32) bool {
+	// Maintain a list of plugins that use hostPrefix
+	pluginsUsingHostPrefix := sets.NewString(string(operv1.NetworkTypeOpenShiftSDN), string(operv1.NetworkTypeOVNKubernetes))
+	if pluginsUsingHostPrefix.Has(networkType) {
+		return true
+	}
+	if hostPrefix != 0 {
```

Comment:
Why do we care what the hostPrefix value is if the networkType is not one of `OpenshiftSDN` or `OVNKubernetes`?

Reply:
> Why do we care what the hostPrefix value is if the networkType is not one of `OpenshiftSDN` or `OVNKubernetes`?

@danwinship suggested that we should still perform the check if the networkType has set the hostPrefix (anything other than a 0) which makes sense since we expect the plugin to follow the OCP rules. This way we give two options to the plugins that don't use this field:

-  keep it unset in which case we won't care or validate,
-  if they set it they should ensure they set it as per our rules.

we can document this to keep it clear.

Reply:
I am not sure I follow. If the plugin is not the one managed by us, why should we care whether the hostPrefix gets used, is set, or not? I think @danwinship 's comment to make it optional makes sense so a field that is not set is not shown in the output of `oc get`

Reply:
> I am not sure I follow. If the plugin is not the one managed by us, why should we care whether the hostPrefix gets used, is set, or not? 

So when we make the field optional and the customer does not specify a value, the default value is zero in which case we skip validation. Initially I had done what you suggested which is to completely skip validation if its not either sdn or ovn. But what Dan suggested was that we should not allow anyone to set _any possible_ value to this field just because its not used which is why I changed it to include the validation if the plugin still sets it.

Basically CNO has a set of checks for hostPrefix field which every plugin should follow. If a plugin does not use it or does not want to follow it, they can unset it. However if they set this field, then they have to ensure they follow CNO's validation rules. See his comment https://github.com/openshift/cluster-network-operator/pull/709#discussion_r453935277.

> I think @danwinship 's comment to make it optional makes sense so a field that is not set is not shown in the output of `oc get`

yea I have made in optional in openshift/api#697 so that it is not shown in the output.